sys/kern/vm_object.c

```diff
@@ -81,13 +81,19 @@ void vm_object_remove_pages(vm_object_t *obj, vm_offset_t off, size_t len) {
 }
 
 #define vm_object_remove_all_pages(obj)                                        \
-  vm_object_remove_pages((obj), 0, (size_t)(-PAGESIZE))
+  vm_object_remove_pages_nolock((obj), 0, (size_t)(-PAGESIZE))
 
-void vm_object_free(vm_object_t *obj) {
-  if (!refcnt_release(&obj->ref_counter))
-    return;
+void vm_object_hold(vm_object_t *obj) {
+  refcnt_acquire(&obj->ref_counter);
+}
+
+void vm_object_drop(vm_object_t *obj) {
+  WITH_MTX_LOCK (&obj->mtx) {
```

Comment:
Is taking the lock necessary? Dropping the refcount is atomic, and if it drops to 0, we're the only thread with a reference to the object, so is synchronization necessary here?

Reply:
Yes, you're right. I misunderstood this.